samples/sample_common/src/vaapi_allocator.cpp

```diff
@@ -206,23 +205,27 @@ mfxStatus vaapiFrameAllocator::AllocImpl(mfxFrameAllocRequest *request, mfxFrame
         if( VA_FOURCC_P208 != va_fourcc )
         {
             unsigned int format;
+            VASurfaceAttrib attrib[2];
```

Comment:
better to make it ```std::vector```

Reply:
why? std::vector is too complicated for this purpose ... VASurfaceAttrib is a small-size struct

Reply:
because you are using explicit indexing and it has potential issue of out of bounds indexing if this code will be changed (for example one more attrib will be added without changing array size).  Using vector + reserve + push_back much more error prone and overhead of heap allocation is negligible